packages/sitecore-jss/src/site/graphql-redirects-service.ts

```diff
@@ -87,13 +90,22 @@ export class GraphQLRedirectsService {
       throw new Error(siteNameError);
     }
 
-    const redirectsResult: Promise<RedirectsQueryResult> = this.graphQLClient.request(this.query, {
-      siteName,
-    });
+    const cacheKey = `redirects-${siteName}`;
+    let data = this.cache.getCacheValue(cacheKey);
 
-    return redirectsResult
-      .then((result: RedirectsQueryResult) => result?.site?.siteInfo?.redirects || [])
-      .catch((e) => Promise.reject(e));
+    if (!data) {
+      // eslint-disable-next-line no-useless-catch
+      try {
+        data = await this.graphQLClient.request<RedirectsQueryResult>(this.query, {
+          siteName,
+        });
+        this.cache.setCacheValue(cacheKey, data);
+      } catch (error) {
+        throw error;
+      }
```

Comment:
The linting was right (I'm looking at `// eslint-disable-next-line no-useless-catch` above). There is no point in having this try / catch.